### PR TITLE
Neighborhood: basic rendering

### PR DESCRIPTION
**Pull request by molly-moen**
This PR adds support for basic rendering of neighborhood maps and adds the plumbing we will need to add and remove paint from the map.

Features included:
- Can deserialize a serialized neighborhood map, which for now must by an 8x8 2d array of objects in the format `{“tileType”:0,“assetId”:18,“value”:0}`, where assetId is optional. `tileType` is an enum used across Maze, we will most likely only user 0 (wall) and 1 (open). `assetId` is used if the tile has any sort of asset on it (trucky, cone, etc.), and it aligns with the neighborhood sprite id mapping defined by curriculum. The value is used if there is a paint bucket on the square, and indicates the number of paint units available.
- Add functions for adding and removing paint, which for now update the `color` field on the cell. This will be expanded on when we add paint glomming.
- Added an `addPegman` function to the mazeController which can be used to add additional pegmen to the maze.

Neighborhood now expects the skin to include an object of `neighborhoodSprites` (see [this PR](https://github.com/code-dot-org/code-dot-org/pull/40543) for initial work on this), as well as an object `sheetRows` in the format: 
```
{
   'other.png': 3,
   'vehicles.png': 7,
    ...
}
```

Where the keys are spritesheet names, and the value is the number of rows in the spritesheet. This is needed for picking out the correct sprite from the spritesheet.  The spritesheets are expected to always have 10 columns.

The display will need some fine-tuning on borders (to be done as a follow-up), but currently looks like this with some hacking on the code-dot-org side:
Screen Shot 2021-05-13 at 11 44 48 AM

## Future Work
- Version up Maze and incorporate these updates with the `code-dot-org` repo.
- Update the `code-dot-org` repo Maze API to use pegman ids and call into paint/neighborhood commands as needed.
- Add any necessary functions to Maze for the management of pegmen in neighborhood--we will definitely need to update reset, because we will need to delete all pegmen on reset. We will also need to add a new function to animationsController for adding a new pegman to the neighborhood.
- Add placeholder pegman capabilities to Maze.

### CODE REVIEW

File: src/neighborhood.js [hunk 1]
```diff
@@ -57,6 +79,45 @@ module.exports = class Neighborhood extends Subtype {
...
+   * @param {String} color Color to paint current location. 
+   *                       Must be hex code or html color.
+  **/ 
+  addPaint(pegmanId, color) {
+    const col = this.maze_.getPegmanX();
+    const row = this.maze_.getPegmanY();
+
+    const cell = this.getCell(row, col);
+    cell.setColor(color);
+    // TODO: update color on map
```
Comment (madelynkasula) [inline src/neighborhood.js:97]: just for my own understanding -- what's the difference between setting the color on the cell and updating the color on the map? why are both actions necessary here?

File: src/neighborhoodCell.js [hunk 2]
```diff
@@ -19,4 +20,34 @@ module.exports = class NeighborhoodCell extends Cell {
...
+   * Serializes this NeighborhoodCell into JSON
+   * @return {Object}
+   * @override
+   */
+  serialize() {
+    return Object.assign({}, super.serialize(), {
+      assetId: this.assetId,
+      hasBucket: this.hasBucket,
+      color: this.color
+    });
```
Comment (madelynkasula) [inline src/neighborhoodCell.js:38]: nit: we can use object destructuring here as a more modern method of merging objects

File: src/neighborhood.js (hunk 1 above)
Comment (molly-moen) [reply src/neighborhood.js:97]: updating the color on the map will involve actually rendering the 'paint' on top of the cell, this is just the data on the cell that we'll use to render the paint. This is mostly a placeholder for now, we are going to do some slightly complicated painting algorithm to make diagonals/lines/individual cells look nice.

File: src/neighborhoodCell.js (hunk 2 above)
Comment (molly-moen) [reply src/neighborhoodCell.js]: updated!

**Review by madelynkasula — APPROVED** [review]
🎨 🏠

**Pull request merged**